crates/plugins/src/lookup.rs

```diff
@@ -33,49 +29,53 @@ impl PluginLookup {
         }
     }
 
-    fn version_string(&self) -> String {
-        self.version
-            .as_ref()
-            .map(|s| s.to_string())
-            .unwrap_or_else(|| String::from("latest"))
-    }
-
-    pub async fn get_manifest_from_repository(&self, plugins_dir: &Path) -> Result<PluginManifest> {
-        log::info!(
-            "Pulling manifest for plugin {} from {}",
-            self.name,
-            SPIN_PLUGINS_REPO
-        );
-        fetch_plugins_repo(plugins_dir, false).await?;
-        let file = File::open(spin_plugins_repo_manifest_path(
-            &self.name,
-            &self.version,
-            plugins_dir,
-        ))
-        .with_context(|| {
-            format!(
-                "{} {} {} in centralized repository",
-                self.name,
-                self.version_string(),
-                PLUGIN_NOT_FOUND_ERROR_MSG,
-            )
+    pub async fn get_manifest_from_repository(
+        &self,
+        plugins_dir: &Path,
+    ) -> PluginLookupResult<PluginManifest> {
+        let url = plugins_repo_url().map_err(Error::UrlParseError)?;
```

Comment:
With the `thiserror` `#[from]` suggestion above:

```suggestion
        let url = plugins_repo_url()?;
```